Code/Tools/Libs/ToolsFoundation/Document/Implementation/DocumentManager.cpp

```diff
@@ -219,6 +220,41 @@ ezStatus ezDocumentManager::CreateOrOpenDocument(bool bCreate, const char* szDoc
   {
     if (DocumentTypes[i]->m_sDocumentTypeName == szDocumentTypeName)
     {
+      // See if there is a default asset document registered for the type, if so clone
+      // it and use that as the new document instead of creating one from scratch.
+      if (bCreate)
+      {
+        ezString ProjectDirectory = ezToolsProject::GetSingleton()->GetProjectDirectory();
+        ezStringBuilder TemplateDocumentPath = ProjectDirectory;
+        ezStringBuilder DocumentFileName;
+        TemplateDocumentPath.AppendPath("DocumentTemplates", szDocumentTypeName);
+        TemplateDocumentPath.ChangeFileExtension("ezDocument");
```

Comment:
Why ".ezDocument" ? Wouldn't it be easier to use the type specific extension, so that one can easily open the document and modify it? Or is it intended to hide the docs from the asset browser? In that case, you could also put them in a folder with the "_data" postfix, to hide them by default.

Reply:
Changed how it works, this is the simple implementation. The idea was to have support for different templates per asset type so that the user can choose in the New > ... context menu for example. Templates are now Default.Extension in the templates folder. This way they can also be edited easily from within the editor.